File: cplusplus/src/base/commons/base-shape.cpp
#include <shapes/base/commons.hpp>

using namespace std;

namespace shapes::base::commons {
BaseShape::BaseShape(const vector<map<string, double>>& dimensions)
  : dimensions(dimensions) { }

BaseShape::~BaseShape() = default;

auto BaseShape::getDimensions() const -> const vector<map<string, double>>& {
    return dimensions;
}

auto BaseShape::showAvailableDimensionCount() const -> size_t {
    return dimensions.size();
}

auto BaseShape::validateShape() -> bool {
    return true;
}
} // namespace shapes::base::commons

File: go/base/commons/base-shape.go
package commons

type BaseShape struct {
	Dimensions []map[string]float64
}

func (BaseShape) New(dimensions []map[string]float64) *BaseShape {
	return &BaseShape{Dimensions: dimensions}
}

func (b *BaseShape) ShowAvailableDimensionCount() int {
	return len(b.Dimensions)
}

func (BaseShape) ValidateShape() bool {
	return true
}
